index.js

```diff
@@ -32,6 +34,10 @@ function gulpAmdCheck(options) {
         file.contents = new Buffer(output.optimizedContent);
       }
 
+      if(options.errorOnUnusedDependencies && output.results.length && output.results[0].unusedDependencies.length) {

```

Comment:
Instead of:
```js
output.results.length && output.results[0].unusedDependencies.length
```
you can use:
```js
output.results.some(function (r) { return r.unusedDependencies.length; })
```
Because it's possible that the first result (`results[0]`) has not any unused deps but the others have.